libcaf_net/caf/net/http/async_client.cpp

```diff
@@ -41,8 +41,13 @@ class async_client_impl : public async_client {
   }
 
   void abort(const caf::error& reason) override {
-    log::net::error("Response abortet with: {}", to_string(reason));
-    response_.set_error(reason);
+    if (response_.valid()) {
+      log::net::error("Response abortet with: {}", to_string(reason));
+      down->shutdown();
+      if (down)
+        down->shutdown();
```

Comment:
This shouldn't call `down->shutdown`, because `abort` is callback for the lower layer to clean up upper layers: https://github.com/actor-framework/actor-framework/blob/main/libcaf_net/caf/net/generic_upper_layer.hpp#L26

Reply:
Refactored.